src/viewport.js

```diff
@@ -366,9 +378,10 @@ export class Viewport {
 
   /** @private */
   resize_() {
-    let oldWidth = this.width_;
-    this.width_ = this.getSize().width;
-    this.changed_(oldWidth != this.width_, 0);
+    let oldSize = this.size_;
+    this.size_ = null;  // Need to recalc.
```

Comment:
I'm not sure I follow. getSize is cached now, so one needs to to bust the cache before reading when it is known that the size changed.